smartcontractkit/external-adapters-js
# Feature/finage v3 uk etf

## Closes #DF-18915

## Description

Add uk_etf endpoint to finage V3. The websocket endpoint does not return a message on initial connection, so adapter may return 504 errors because it did not receive a WS message. After the adapter receives the first ws message from finage, it'll correctly return an answer.

## Changes
- Add uk_etf endpoint for https
- Add uk_etf endpoint for websocket
- Add test for both endpoints
- Cleaned up tests (removed unnecessary parameters and reduced code redundancy)
- Fix spelling error (symbool -> symbol)



## Steps to Test

1. To test https endpoint, set `WS_ENABLED=false`, and `yarn build && yarn start`
2. Send a query
```
curl --location 'localhost:8080' \
--header 'Content-Type: application/json' \
--data '{
    "id": "123",
    "data": {
        "from": "cspx",
        "endpoint": "uk_etf"
    }
}'
```
4. Check the result is correct
```
{
    "data": {
        "result": 418
    },
    "result": 418,
    "timestamps": {
        "providerDataRequestedUnixMs": 1685973991459,
        "providerDataReceivedUnixMs": 1685973992451,
        "providerIndicatedTimeUnixMs": 1685973992314
    },
    "statusCode": 200,
    "meta": {
        "adapterName": "FINAGE",
        "metrics": {
            "feedId": "{\"base\":\"cspx\"}"
        }
    }
}
```
5. Test websocket endpoint, by setting `WS_ENABLED=true` and sending request from step 2
6. Check the responses. The endpoint only works after the adapter receives it's first ws message. So it may not return a response at the start or during off market hours. 
```
{
    "data": {
        "result": 446.06
    },
    "result": 446.06,
    "timestamps": {
        "providerDataStreamEstablishedUnixMs": 1685974153485,
        "providerDataReceivedUnixMs": 1685974435721,
        "providerIndicatedTimeUnixMs": 1685973509000
    },
    "statusCode": 200,
    "meta": {
        "adapterName": "FINAGE",
        "metrics": {
            "feedId": "{\"base\":\"cspx\"}"
        }
    }
}
```
## Quality Assurance
